aws-janitor/resources/efs.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/efs"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	maxRetries   = 120
+	pollInterval = time.Second
+)
+
+// ElasticFileSystems: https://docs.aws.amazon.com/sdk-for-go/api/service/efs/#EFS.DescribeFileSystems
+
+type ElasticFileSystems struct{}
+
+func (ElasticFileSystems) MarkAndSweep(opts Options, set *Set) error {
```

Comment:
per the lint error, it might be worth breaking this function up a bit - maybe the handling of mount targets should be a separate function?

(that said, I probably need to tune that linter a bit.)

Reply:
It's reasonable; I felt this function was getting too large as I was writing it, but I didn't like the alternative of passing everything as arguments to helper functions.

It would be nice to make things like the EFS client and logger struct fields so we don't have to pass everything around as args, but that requires a bit more refactoring (making constructors for all of these types). If you have any other ideas for working around this let me know.

Reply:
I split this into a separate function. I had thought about doing this for each stage of `MarkAndSweep()`. I could proceed with this, but I was reluctant because the "style" would stray from what we're doing for other resource types.